src/util/git.js

```diff
@@ -360,14 +360,15 @@ export default class Git {
 
   async setRef(refs: GitRefs): Promise<string> {
     // get commit ref
-    const {hash} = this;
+    const {hash, cwd} = this;
 
     const names = Object.keys(refs);
 
     if (Git.isCommitHash(hash)) {
       for (const name in refs) {
         if (refs[name] === hash) {
           this.ref = name;
+          await child.spawn('git', ['fetch', 'origin', name], {cwd});
```

Comment:
`git ls-remote` isn't enough to make the files in the pull request available, so we need to fetch the reference here.